lib/charms/smtp_integrator/v0/smtp.py

```diff
@@ -178,6 +186,12 @@ def password(self) -> str:
         assert self.relation.app
         return self.relation.data[self.relation.app].get("password")
 
+    @property
+    def password_id(self) -> str:
+        """Fetch the SMTP password from the relation."""
+        assert self.relation.app
```

Comment:
Need not be addressed now, but assert statements are not effective when Python is run with optimisation and are mainly used for debugging/dev purposes (https://docs.python.org/3.10/reference/simple_stmts.html#the-assert-statement). Is that the intention here?

Reply:
The assert was added so that mypy acknowledges that  `self.relation.app` is not None in the next statement